### PR TITLE
feat: Adding Sink trait impl to Either

### LINKED ISSUE #6181: Add Sink trait impl to Either
**Is your feature request related to a problem? Please describe.**

The docs of `tokio_util::either::Either` state

> Combines two different futures, streams, or sinks having the same associated types into a single type.

Though Either does not implement Sink

**Describe the solution you'd like**
Implement Sink for Either if Error types and message type match, similar to Stream

https://docs.rs/tokio-util/0.7.10/tokio_util/either/enum.Either.html#impl-Stream-for-Either%3CL,+R%3E

### PR DESCRIPTION
**Pull request by mobley-trent**
Closes #6181

### CODE REVIEW

File: tokio-util/src/either.rs [hunk 1]
```diff
@@ -164,6 +164,31 @@ where
     }
 }
 
+impl<L, R, Item> futures_sink::Sink<Item> for Either<L, R>
+where
+    L: futures_sink::Sink<Item>,
+    R: futures_sink::Sink<Item>,
+{
+    type Error = L::Error;
```
Comment (Darksonn) [inline tokio-util/src/either.rs:172]:
```suggestion
impl<L, R, Item, Error> futures_sink::Sink<Item> for Either<L, R>
where
    L: futures_sink::Sink<Item, Error = Error>,
    R: futures_sink::Sink<Item, Error = Error>,
{
    type Error = Error;
```

File: tokio-util/src/either.rs [hunk 2]
```diff
@@ -164,6 +164,31 @@ where
...
 }
 
+impl<L, R, Item, Error> futures_sink::Sink<Item> for Either<L, R>
+where
+    L: futures_sink::Sink<Item, Error = Error>,
+    R: futures_sink::Sink<Item, Error = Error>,
+{
+    type Error = Error;
+
+    fn poll_ready(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<()>> {
```
Comment (Darksonn) [inline tokio-util/src/either.rs:174]: In each method, you need this:
```suggestion
    fn poll_ready(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), Error>> {
```
Right now, you're using `std::io::Error` instead, which is the wrong error type.

**Comment by mobley-trent** [comment]
Hey @Darksonn should the return type be `Result<(), Error>` or `Result<(), Self::Error>` ?
Also I think `Error` should be instantiated to `L::Error` instead of just error. What are your thoughts ?

File: tokio-util/src/either.rs (hunk 2 above)
Comment (mobley-trent) [reply tokio-util/src/either.rs]: How do I fix it ?

**Comment by Darksonn** [comment]
All of `Error`, `Self::Error`, `L::Error`, `R::Error` refer to the same type. I would prefer that you use the first one.

You probably have to explicitly say `std::result::Result` because the `std::io::Result` type alias does not seem to allow you to specify the error type.

**Review by Darksonn — APPROVED** [review]
Thank you.

**Pull request merged**